frontend/src/homePage.js

```diff
@@ -0,0 +1,76 @@
+import React, { useState, Component } from "react";
+import ReactDOM from "react-dom";
+import NavBar from "./homePage/NavBar";
+import LandingPage from "./homePage/LandingPage";
+import About from "./homePage/About";
+
+class Home extends React.Component {
+  constructor(props) {
+    super(props);
+    this.loginClick = this.handleLoginClick.bind(this);
+    this.folderClick = this.handleFoldersClick.bind(this);
+    this.state = {
+      userId: "null",
+      loginStatus: false,
+      logoutUrl: "null",
+      isDataFetched: false,
+    };
+    this.checkLogin();
+  }
+
+  // Calls the login Servlet when the page loads for the first time
+  checkLogin(e) {
+    const userInfo = fetch("/login")
+      .then((response) => response.json())
+      .then((info) => {
+        if (info["showTabStatus"] === true) {
+          this.setState({ loginStatus: true, logoutUrl: info["logoutUrl"], isDataFetched: true, userId: info['userInfo']['userId'] });
+        } else {
+          this.setState({ loginStatus: false, isDataFetched: true });
+        }
+      });
+  }
+
+  // Handles buttons that login user
+  handleLoginClick = (e) => {
+    const userInfo = fetch("/login")
+      .then((response) => response.json())
+      .then((info) => {
+        // Check if the user has logged in before
+        if (info["showTabStatus"] === true) {
+          window.location = this.state.logoutUrl;
+        } else {
+          window.location = info["loginUrl"];
+        }
+      });
+  };
+
+
+  handleFoldersClick = (e) => {
+    console.log("rerouting to folders page");
```

Comment:
We will probably have to take this out later on, but I think it's fine to leave it for now.